### PR TITLE
16577 - Alteration - Name Request changes

### PR DESCRIPTION
**Pull request by seeker25**
*Issue #:* /bcgov/entity#16577

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Review by Jxio — APPROVED** [review]
LGTM

File: src/components/common/YourCompany/ChangeBusinessType.vue [hunk 1]
```diff
@@ -320,12 +344,27 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
...
+    // Named companies to CC or ULC require a name request.
+    if (this.isCommunityContribution || this.isUnlimitedLiability) {
+      return true
+    }
+    // Named ULC to BC Limited require a name request.
+    if (this.getEntitySnapshot?.businessInfo?.legalType === CorpTypeCd.BC_ULC_COMPANY && this.isBcLimited) {
+      return true
+    }
+    return false
+  }
```
Comment (severinbeauvais) [inline src/components/common/YourCompany/ChangeBusinessType.vue:360]: Nice and clean (easy to review) 👍

**Review by severinbeauvais — APPROVED** [review]

**Review by JazzarKarim — APPROVED** [review]
LGTM Travis!

**Comment by seeker25** [comment]
Thank you both

**Comment by seeker25** [comment]
image

image

File: src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue [hunk 2]
```diff
@@ -225,11 +230,39 @@ export default class CorrectNameRequest extends Mixins(CommonMixin, NameRequestM
...
+  /* Generate content of error depending on name request type. */
+  nameRequestErrorText (nr: NrResponseIF): string {
+    const isConversionOrAlterationNameRequest = nr.request_action_cd === NrRequestActionCodes.CONVERSION
+    let dialogContent = ''
+    if (isConversionOrAlterationNameRequest) {
+      dialogContent = `<p class="info-text">This alteration name request from ` +
+        `${GetCorpFullDescription(nr.entity_type_cd)} to ${GetCorpFullDescription(nr.legalType)} ` +
+        `does not match the current business type ` +
+        `<b>${GetCorpFullDescription(this.getEntityType)}</b>.\n\n` +
+        `The Name Request type must match the business type before you can continue.</p>`
```
[the remark this answers is not in the document]
Comment (seeker25) [reply src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue:251]: I'll try this one in the next PR - after QA

[the remark this answers is not in the document]
Comment (severinbeauvais) [reply src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue:251]: Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

**Pull request merged**